python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-from dataclasses import dataclass, field
 from datetime import datetime
-from enum import Enum, auto
-from typing import Any
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, Field
 
 DATAHUB_DATE_FORMAT = "%Y%m%d"
 
 
-@dataclass
-class CatalogueMetadata:
-    name: str
-    description: str
-    owner: str
-    tags: list[str] = field(default_factory=list)
+class DatabaseStatus(Enum):
+    PROD = "PROD"
+    PREPROD = "PREPROD"
+    DEV = "DEV"
+
 
+class RelationshipType(Enum):
+    PARENT = "PARENT"
+    PLATFORM = "PLATFORM"
```

Comment:
Why have these enum values changed from [Enum.auto()](https://docs.python.org/3/library/enum.html#enum.auto)?

Reply:
when `model_dump_json` is called on a pydantic model eg to encode classes to pass to datahub, enum classes are converted to their string representations. With auto(), this is an integer value which can't be read as it's actual value.

Reply:
If we're above python 3.10, we can swap `(Enum)` to `(StrEnum)`, and we should get compatible behaviour for both Enum and Pydantic.
>Note Using [auto](https://docs.python.org/3/library/enum.html#enum.auto) with [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) results in the lower-cased member name as the value.

Reply:
If we drop support for python 3.10, we could maybe use StrEnum with Auto https://docs.python.org/3/library/enum.html#enum.StrEnum